src/runtime_src/core/pcie/tools/xbmgmt/cmd_hotplug.cpp

```diff
@@ -0,0 +1,197 @@
+            break;
+
+        switch (opt) 
+        {
+            case '0':
+		char ch;
+                index = bdf2index(optarg);
+                if (index == UINT_MAX)
+                    return -ENOENT;
+            	std::cout << "CAUTION: Performing hot removal. This command is going to impact both user pf and mgmt pf. " <<
+                "Please make sure no application is currently running." << std::endl;
+		std::cout << "Proceed[y/n]? " << std::endl;
+		std::cin >> ch;
+		if (ch == 'y' || ch == 'Y')
+			break;
+		else
+			return -EINVAL;
+
+            default:
+                return -EINVAL;
+        } 
+    }
+
+    /* Remove user_pf */
+    auto uDev = pcidev::get_dev(index, true);
+    removeDevice(uDev);
+
+    /* Remove mgmt_pf */
+    auto mDev = pcidev::get_dev(index, false);
+    removeDevice(mDev);
+    
+    return 0;
+}
+
+static int removeDevice(std::shared_ptr<pcidev::pci_device> dev)
+{
+    std::string sysfs_path = dev->get_sysfs_path("", "remove");
+
+    if (sysfs_path.empty()) {
+        return -ENOENT;
+    }
+
+    std::ofstream ofile(sysfs_path);
+
+    if (!ofile.is_open()) {
+        std::cout << "Failed to open " << sysfs_path << ":" << strerror(errno) << std::endl;
+        return -ENOENT;
+    }
+
+    /* "echo 1 > /sys/bus/pci/<EndPoint>/remove" to trigger hot remove of the device */
+    ofile << 1;
+    ofile.flush();
+    if (!ofile.good()) {
+        std::cout << "Failed to write " << sysfs_path << ":"  << strerror(errno) << std::endl;
+        ofile.close();
+        return -EINVAL;
+    }
+
+    ofile.close();
+
+    return 0;
+}
+
+
+const char *subCmdHpRescanDesc = "Perform hot rescan on the device";
+const char *subCmdHpRescanUsage = "(no options supported)";
+
+int hpRescanHandler(int argc, char *argv[])
+{
+    sudoOrDie();
+
+    if (argc != 1)
+        return -EINVAL;
+
+    const std::string sysfs_path = findRootPort() + "/rescan";
+
+    if (sysfs_path.empty()) {
+        return -ENOENT;
+    }
+
+    std::ofstream ofile(sysfs_path);    
+    
+    if (!ofile.is_open()) {
+        std::cout << "Failed to open " << sysfs_path << ":" << strerror(errno) << std::endl;
+        return -ENOENT;
+    }
+   
+    /* "echo 1 > /sys/bus/pci/<Root Port>/rescan" to trigger the rescan for hot plug devices */ 
+    ofile << 1;
+    ofile.flush();
+    if (!ofile.good()) {
+        std::cout << "Failed to write " << sysfs_path << ":"  << strerror(errno) << std::endl;
+        ofile.close();
+        return -EINVAL;
+    }
+
+    ofile.close();
+
+    return 0;
+}
+
+static std::string findRootPort(void)
+{
+    std::string rootPortPath = "";
+    path fPath(SYSFS_PATH);
+    
+    for (auto file = directory_iterator(fPath); file != directory_iterator(); file++)
+    {
+        rootPortPath = file->path().string();
+        for (auto jfile = directory_iterator(rootPortPath); jfile != directory_iterator(); jfile++)
+        {
+            std::string dirName(jfile->path().string());
```

Comment:
`boost::filesystem::path dir = jfile->path() `

Reply:
Added this change. Also added boost::filesystem for other cases instead of std::string. Please Review.